s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -267,7 +303,7 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
 
     val headers: S3Headers = S3Headers(sse.fold[Seq[HttpHeader]](Seq.empty) { _.headersFor(UploadPart) })
 
-    SplitAfterSize(chunkSize)(Flow.apply[ByteString])
+    SplitAfterSize(chunkSize)(conditionalAppendFl)
```

Comment:
seeing this makes me think this may just be a bug in [`SplitAfterSize`](https://github.com/eyalfa/alpakka/blob/11eacfccf6d414dbf9c195b7b33150262325c3a8/s3/src/main/scala/akka/stream/alpakka/s3/impl/SplitAfterSize.scala#L38). I don't think it creates a single empty sub-stream when seeing a completely empty source stream. Perhaps `onUpstreamFinish()` should be overridden to emit `NewStream` if nothing else was seen?

Reply:
not sure, it does make sense that empty input yields empty output, in any case this is use-case specific.